base_layer/core/src/transactions/transaction_builder/transaction_builder.rs

```diff
@@ -0,0 +1,800 @@
+            lock_height: 0,
+            kernel_features: KernelFeatures::empty(),
+            burn_commitment: None,
+            own_address,
+        })
+    }
+
+    /// Set the fee per weight for the transaction. See (Fee::calculate)[Struct.Fee.html#calculate] for how the
+    /// absolute fee is calculated from the fee-per-gram value. This will take precedence over the fee set by
+    /// `with_fee`.
+    pub fn with_fee_per_gram(&mut self, fee_per_gram: MicroMinotari) -> &mut Self {
+        self.fee_per_gram = Some(fee_per_gram);
+        self
+    }
+
+    pub fn with_lock_height(&mut self, lock_height: u64) -> &mut Self {
+        self.lock_height = lock_height;
+        self
+    }
+
+    /// Sets the fee of the transaction. Fee per gram takes precedence over this value.
+    pub fn with_fee(&mut self, fee: MicroMinotari) -> &mut Self {
+        self.fee = fee;
+        self
+    }
+
+    /// Sets the transaction type, default is TxType::PaymentToOther
+    pub fn with_tx_type(&mut self, tx_type: TxType) -> &mut Self {
+        self.tx_type = tx_type;
+        self
+    }
+
+    /// Sets the payment id for the transaction. This is used to identify the transaction and is included in the
+    /// transaction metadata.
+    pub fn with_payment_id(&mut self, payment_id: MemoField) -> &mut Self {
+        self.payment_id = Some(payment_id);
+        self
+    }
+
+    pub fn with_kernel_features(&mut self, kernel_features: KernelFeatures) -> &mut Self {
+        self.kernel_features = kernel_features;
+        self
+    }
+
+    /// Add a recipient to the transaction.
+    pub async fn add_recipient(
+        &mut self,
+        recipient_address: TariAddress,
+        recipient_output: WalletOutput,
+        sender_offset_key_id: Option<TariKeyId>,
+    ) -> Result<&mut Self, TransactionBuilderError> {
+        let kernel_nonce = self
+            .key_manager
+            .get_next_key(TransactionKeyManagerBranch::KernelNonce.get_branch_key())
+            .await?;
+        let recipient_output = OutputPair::new(recipient_output, kernel_nonce.key_id, sender_offset_key_id);
+        let recipient_details = RecipientDetails {
+            output: recipient_output,
+            recipient_address,
+        };
+        self.recipients.push(recipient_details);
+        Ok(self)
+    }
+
+    pub async fn with_input(&mut self, input: WalletOutput) -> Result<&mut Self, TransactionBuilderError> {
+        let nonce = self
+            .key_manager
+            .get_next_key(TransactionKeyManagerBranch::KernelNonce.get_branch_key())
+            .await?;
+        let pair = OutputPair::new(input, nonce.key_id, None);
+        self.inputs.push(pair);
+        Ok(self)
+    }
+
+    /// This will allow the receipient to sign the burn commitment
+    pub fn with_burn_commitment(&mut self, commitment: Option<CompressedCommitment>) -> &mut Self {
+        self.burn_commitment = commitment;
+        self
+    }
+
+    pub async fn with_output(
```

Comment:
Otherewise it is not clear when one reads the code
```suggestion
    pub async fn with_sender_custom_output(
```

Reply:
nope I disagree, its not forced to a sender output, like it was previously